### PR TITLE
Remove global project cache and use interfaces for admission plugin config

### PR DESCRIPTION
**Pull request by csrwng**
Follow-up to #6572 
Adds interfaces that plugins implement to request objects on initialization

### CODE REVIEW

**Comment by csrwng** [comment]
@deads2k @stevekuznetsov

**Comment by stevekuznetsov** [comment]
@csrwng thanks for picking this up. I'll close https://github.com/openshift/origin/pull/6572

**Comment by csrwng** [comment]
rebased and added additional interface to validate that initialization has taken place

File: pkg/build/admission/admission.go [hunk 1]
```diff
@@ -31,12 +27,14 @@ type buildByStrategy struct {
 	client client.Interface
 }
 
+var _ = oadmission.WantsOpenshiftClient(&buildByStrategy{})
+var _ = oadmission.Validater(&buildByStrategy{})
```
Comment (deads2k) [inline pkg/build/admission/admission.go]: `Validator` please

Comment (csrwng) [inline pkg/build/admission/admission.go]: fixed

**Comment by deads2k** [comment]
lgtm [merge]

**Comment by csrwng** [comment]
Now passing project cache to BuildKubernetesMasterConfig

**Comment by csrwng** [comment]
failed on new-app timeout 
[merge]

**Pull request merged**